homeassistant/components/keenetic_ndms2/config_flow.py

```diff
@@ -0,0 +1,145 @@
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+        if user_input is not None:
+            for entry in self.hass.config_entries.async_entries(DOMAIN):
+                if entry.data[CONF_HOST] == user_input[CONF_HOST]:
+                    return self.async_abort(reason="already_configured")
+
+            _client = Client(
+                TelnetConnection(
+                    user_input[CONF_HOST],
+                    user_input[CONF_PORT],
+                    user_input[CONF_USERNAME],
+                    user_input[CONF_PASSWORD],
+                )
+            )
+
+            try:
+                router_info = await self.hass.async_add_executor_job(
+                    _client.get_router_info
+                )
+
+                return self.async_create_entry(title=router_info.name, data=user_input)
+            except ConnectionException:
+                errors["base"] = "cannot_connect"
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST): str,
+                    vol.Required(CONF_USERNAME): str,
+                    vol.Required(CONF_PASSWORD): str,
+                    vol.Optional(CONF_PORT, default=DEFAULT_TELNET_PORT): int,
+                }
+            ),
+            errors=errors,
+        )
+
+
+class KeeneticOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle options."""
+
+    def __init__(self, config_entry):
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, _user_input=None):
+        """Manage the options."""
+        return await self.async_step_device_tracker()
+
+    async def async_step_device_tracker(self, user_input=None):
+        """Manage the device tracker options."""
+        if user_input is not None:
+            return self.async_create_entry(
+                title="",
+                data={
+                    **user_input,
+                    CONF_INTERFACES: [
+                        interface.strip()
+                        for interface in user_input[CONF_INTERFACES].split(",")
+                    ],
+                },
+            )
+
+        options = vol.Schema(
+            {
+                vol.Optional(
+                    CONF_SCAN_INTERVAL,
+                    default=self.config_entry.options.get(
+                        CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+                    ),
+                ): int,
+                vol.Optional(
+                    CONF_CONSIDER_HOME,
+                    default=self.config_entry.options.get(
+                        CONF_CONSIDER_HOME, DEFAULT_CONSIDER_HOME
+                    ),
+                ): int,
+                vol.Optional(
+                    CONF_INTERFACES,
+                    default=", ".join(
```

Comment:
Isn't it better to use a multi select form for this option?

https://github.com/home-assistant/core/blob/82607977efaa385e6097509ada5a1153f95214b2/homeassistant/helpers/config_validation.py#L693-L694

Reply:
`multi_select` doe not fit here since validates against a predefined/known set of values.
But `ensure_list_csv` first.
Also, those are validators/parsers and won't help with the default value building

Reply:
Can't we list the available interfaces?

Reply:
Not yet.

Reply:
Ok.